ParseUI-Widget/src/main/java/com/parse/ParseQueryAdapter.java

```diff
@@ -725,11 +661,12 @@ private View getDefaultView(Context context) {
   }
 
   private int getPaginationCellRow() {
-    return objects.size();
+    return getObjects().size();
   }
 
   private boolean shouldShowPaginationCell() {
-    return paginationEnabled && objects.size() > 0 && hasNextPage;
+    ParseQueryPager<T> pager = getPager();
+    return paginationEnabled && pager.getObjects().size() > 0 && pager.hasNextPage();
```

Comment:
I'm going to leave it so that `getObjects()` and `hasNextPage()` are ensured to be called on the same instance of `pager`